src/julia.h

```diff
@@ -397,6 +397,7 @@ typedef struct {
     unsigned exportp:1;
     unsigned imported:1;
     unsigned deprecated:1;
+    jl_sym_t *deprecation_preferred_name;  // non-NULL when new type incompatible with old
```

Comment:
In light of such concerns, it's worth pointing out that an alternative to this approach is to have separate deprecations for the type and for the constructor. (Don't know whether that would be better, just thought I'd mention it.)